src/lib/TCPSOCKET/tcpsocket.h

```diff
@@ -1,51 +1,39 @@
 #pragma once
 #if defined(TARGET_RX)
 
-#include <cstdint>
-#include <cstring>
-#include "ESPAsyncWebServer.h"
+#if defined(PLATFORM_ESP8266)
+#include "ESPAsyncTCP.h"
+#else
+#include "AsyncTCP.h"
+#endif
+#include "CRSFConnector.h"
 #include "crsf2msp.h"
 #include "msp2crsf.h"
-#include "FIFO.h"
 
-#define BUFFER_OUTPUT_SIZE 1024
-#define BUFFER_INPUT_SIZE 1024
+class TCPSOCKET final : public CRSFConnector
+{
+public:
+    TCPSOCKET();
+    void begin();
 
-// buffers reads and write to the specified TCP port
+    void forwardMessage(const crsf_header_t *message) override;
 
-class TCPSOCKET
-{
 private:
-    static TCPSOCKET *instance;
-    //std::vector<AsyncClient *> clients; // a list to hold all clients for MSP to WIFI bridge nut not using multi-client for now
-
     AsyncServer *TCPserver = nullptr;
     AsyncClient *TCPclient = nullptr;
-    static const uint32_t clientTimeoutS = 2U;
+    CROSSFIRE2MSP crsf2msp;
+    MSP2CROSSFIRE msp2crsf;
 
```

Comment:
These were dynamically allocated for a reason. CROSSFIRE2MSP is over 512bytes. They should only be created on first client connect. ESP8285 needs that RAM.